tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,616 @@
+
+		getEndTime := func() time.Time { return endTime }
+		noOpModifier := func() {}
+
+		queries := []string{
+			fmt.Sprintf(`container_memory_usage_bytes{pod="%s",container="nginx-gateway"}`, ngfPodName),
+			// We don't need to check all nginx_gateway_fabric_* metrics, as they are collected at the same time
+			fmt.Sprintf(`nginx_gateway_fabric_nginx_reloads_total{pod="%s"}`, ngfPodName),
+		}
+
+		for _, q := range queries {
+			Eventually(
+				framework.CreateMetricExistChecker(
+					promInstance,
+					q,
+					getEndTime,
+					noOpModifier,
+				),
+			).WithTimeout(metricExistTimeout).WithPolling(metricExistPolling).Should(Succeed())
+		}
+
+		checkContainerLogsForErrors(ngfPodName, false)
+
+		reloadCount, err := framework.GetReloadCount(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		reloadAvgTime, err := framework.GetReloadAvgTime(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		reloadBuckets, err := framework.GetReloadBuckets(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsCount, err := framework.GetEventsCount(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsAvgTime, err := framework.GetEventsAvgTime(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsBuckets, err := framework.GetEventsBuckets(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+			Container: "nginx-gateway",
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		timeToReadyTotal, err := calculateTimeToReadyTotal(logs, timeToReadyStartingLogSubstring)
+		Expect(err).ToNot(HaveOccurred())
+
+		timeToReadyAvgSingle, err := calculateTimeToReadyAverage(logs)
+		Expect(err).ToNot(HaveOccurred())
+
+		results := reconfigTestResults{
+			Name:                 testName,
+			EventsBuckets:        eventsBuckets,
+			ReloadBuckets:        reloadBuckets,
+			NumResources:         resourceCount,
+			TimeToReadyTotal:     timeToReadyTotal,
+			TimeToReadyAvgSingle: timeToReadyAvgSingle,
+			NGINXReloads:         int(reloadCount),
+			NGINXReloadAvgTime:   int(reloadAvgTime),
+			EventsCount:          int(eventsCount),
+			EventsAvgTime:        int(eventsAvgTime),
+		}
+
+		err = writeReconfigResults(outFile, results)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	// Test 1 - Resources exist before start-up
+	It("test 1 - 30 resources", func() {
+		resourceCount := 30
+		timeToReadyStartingLogSubstring := "Starting NGINX Gateway Fabric"
+		test := createResourcesGWLast(resourceCount)
+
+		Expect(test).To(Succeed())
+		Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+		ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+		collectMetrics("1",
+			resourceCount,
+			timeToReadyStartingLogSubstring,
+			ngfPodName,
+			startTime,
+		)
+	})
+
+	It("test 1 - 150 resources", func() {
```

Comment:
Can we name these tests a bit better, following the more sentence-like structure that we use elsewhere?

Reply:
Yea sure, I can't decide what would be a good fit, what do you think of the following options:

- `It("runs test 1 with 150 resources")`
- `It("passes test 1 with 150 resources")`
- `It("collects metrics on test 1 with 150 resources")`
- `It("runs the reconfiguration test where resources exist before startup with 150 resources")`

Reply:
Ultimately we're just gathering metrics, right? So maybe something like `It("gathers metrics where 150 resources exist before startup")`.

Reply:
Okay I could see that for test 1, but what would you recommend for the other test cases which are a little more difficult to phrase:

> Test 2 - Start NGF, deploy Gateway, create many resources attached to GW

`It("gathers metrics where NGF is deployed, we deploy a Gateway resource, then create 150 resources attached to the Gateway")`? I can't think of a clear and helpful message when I try to explain the test.

Reply:
You could utilize `When()` or `Context()` blocks on top of the `It()` blocks to help with some of this, and then combine common tests under those blocks.

```
When("NGF and Gateway resource are deployed first", func(){
       It("gathers metrics after creating 150 resources"
})
```

Something like that.

Reply:
Ah okay that helps a lot, thanks, I'll commit something for review

Reply:
committed the changes, I will comment above on the DescribeTable with Entries review feedback to see if that is still beneficial given the new changes.

Reply:
I think the new setup is good!